sycl/include/CL/sycl/define_vendors.hpp

```diff
@@ -0,0 +1,13 @@
+//==---------- define_vendors.hpp ----- Preprocessor directives ------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#define SYCL_IMPLEMENTATION_ONEAPI
+#define SYCL_FEATURE_SET_FULL
+#define SYCL_IMPLEMENTATION_INTEL
```

Comment:
I believe we need to document the macro in https://github.com/intel/llvm/blob/sycl/sycl/doc/PreprocessorMacros.md

Reply:
I'm not sure about this.  I thought that document described the preprocessor macros that the **user** can set, which have an effect on the compiler.  However, the user is not supposed to set macros like `SYCL_IMPLEMENTATION_INTEL`.  Rather, the implementation sets the macros, and these user can only test them.

Reply:
I think we should have two documents: for macros, that user defines, and for macros, that compiler defines, but that’s a job for a separate PR.